src/modules/party-external-rating/party-external-rating.controller.ts

```diff
@@ -0,0 +1,48 @@
+import { Controller, Get, Inject, Param } from '@nestjs/common';
+import { ApiInternalServerErrorResponse, ApiNotFoundResponse, ApiOkResponse, ApiOperation, ApiParam } from '@nestjs/swagger';
+
+import { GetPartyExternalRatingResponse, GetPartyExternalRatingsResponse } from './dto/get-party-external-ratings-response.dto';
+import { PARTY_EXTERNAL_RATINGS_PROVIDER_SYMBOL } from './party-external-rating.module-definition';
+import { PartyExternalRatingsProvider } from './party-external-ratings.provider';
+
+@Controller()
+export class PartyExternalRatingController {
+  constructor(@Inject(PARTY_EXTERNAL_RATINGS_PROVIDER_SYMBOL) private readonly partyExternalRatingsProvider: PartyExternalRatingsProvider) {}
+
+  @Get('/party/:partyIdentifier/external-rating')
+  @ApiOperation({ summary: 'Get all external ratings for a party.' })
+  @ApiOkResponse({
+    description: 'The external ratings for the party have been successfully retrieved.',
+    type: GetPartyExternalRatingResponse,
+    isArray: true,
+  })
+  @ApiNotFoundResponse({
+    description: 'The specified party, or the external ratings for that party, were not found.',
+  })
+  @ApiInternalServerErrorResponse({
+    description: 'An internal server error has occurred.',
+  })
+  async getExternalRatingsForParty(
```

Comment:
can we add above
```
  @ApiParam({
    name: 'partyIdentifier',
    required: true,
    type: 'string',
    description: 'Description here',
    example: 'Example here',
  })
```

Reply:
Just to confirm -- this decorator only provides documentation (for Swagger), it doesn't actually perform any validation when the application's running?

Reply:
Also, if that's the case, should we add some `class-validator` decorators to the DTO TypeScript classes? (Like `@IsString()`, `@IsNotEmpty()`?)

Reply:
That's right, the decorator from above is only for documentation. And yes, we should add the validation as well through a DTO

Reply:
@catenoid can I just double-check - are there any classes in this PR you were talking about adding `class-validator` decorators to? Or were you just asking in general?

Reply:
I think for that comment, I was looking at the classes in `src/modules/party-external-rating/dto/get-party-external-ratings-response.dto.ts`. From what I can tell, the `@ApiResponseProperty` is Swagger specific, and so doesn't provide input validation?

Reply:
Thanks @catenoid - `get-party-external-ratings-response.dto.ts` is only used as a response object, not a request object, so I don't think we need any validation on it